crates/http/src/spin.rs

```diff
@@ -114,7 +114,9 @@ impl SpinHttpExecutor {
         .await??;
 
         let mut response = http::Response::builder().status(res.status);
-        Self::append_headers(response.headers_mut().unwrap(), res.headers)?;
+        if let Some(headers) = response.headers_mut() {
```

Comment:
Has this actually ever failed?
Theoretically, we just created a new empty response, so the `headers_mut()` call only ever fail if the builder has an error — https://docs.rs/http/latest/http/response/struct.Builder.html#method.headers_mut


That being said, I don't have anything against an extra check here.

Reply:
I was getting a panic when not setting the headers with the go sdk

Reply:
I think If the wit says optional we should implement it as defined

Reply:
But I don't think this is the right fix.
This checks that _the new Rust HTTP response we just created with the builder_ doesn't have an empty header map.

I'd love to have a look at the actual panic.

Reply:
Looks like I only get it when there is no status code and no headers.  Status code is required so an error is correct.  It just should be more useful.  I'll take another look

```
2022-04-18T16:25:36.837707Z TRACE prepare_component{component="tinygo-hello" env=None args=None}: spin_engine: Creating store.
thread 'tokio-runtime-worker' panicked at 'called `Option::unwrap()` on a `None` value', crates/http/src/spin.rs:117:53
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Reply:
Ah, I assume this is because `res.status` is 0 when it's not set, which results in an error there.

Could you check for `res.status != 0` and for `None` on headers?

Thanks!

Reply:
Can you check if this is what you meant?

Reply:
Never mind I didn't refresh